### PR TITLE
server: Implement supply function after validation

### LINKED ISSUE #64: Implement an interface that executes functions when all of the validations are satisfied
## Feature Request

**Is your feature request related to a problem? Please describe.**  

Currently, the server creates a value beforehand and returns the generated value.
We need to create a more lazy return value.

**Describe the solution you'd like**  

Implement an interface that executes functions when all of the validations are satisfied

**Describe alternatives you've considered**  

**Additional context**

### PR DESCRIPTION
**Pull request by HyeockJinKim**
# Summary 
Implement the function to execute supplier
when the condition is satisfied.

- It closes #64

## How to Review? 

- Please refer to my self-review submitted as the review.

### CODE REVIEW

File: packages/server/src/index.ts [hunk 1]
```diff
@@ -38,8 +38,9 @@ type RegisterInfo = {
 };
 
 type SubmitInfo<T> = {
-  validate: KeySupplier<string, Predicate<string>, SubmitInfo<T>>;
+  validate: KeySupplier<string, Predicate<string>, SubmitInfo<T>;
   confirm: Supplier<T, T | null>;
+  supply: Supplier<() => T, T | null>;
```
Comment (HyeockJinKim) [inline packages/server/src/index.ts:43]: Takes a function to be executed as an argument.

File: packages/server/src/index.ts [hunk 2]
```diff
@@ -166,6 +167,12 @@ export function verifyProperty<T>(
       }
       return response;
     },
+    supply(supplier: () => T): T | null {
+      if (!isPassed) {
+        return null;
+      }
+      return supplier();
+    },
```
Comment (HyeockJinKim) [inline packages/server/src/index.ts:175]: implementation is simple.

File: packages/server/test/auth.spec.ts [hunk 3]
```diff
@@ -1,5 +1,7 @@
 import { expect } from 'chai';
-import { AuthType, hashProperty, KeyType, objectToProperty, Property, PropertyType } from '@0auth/message';
+import {
+  AuthType, hashProperty, KeyType, objectToProperty, Property, PropertyType,
+} from '@0auth/message';
```
Comment (HyeockJinKim) [inline packages/server/test/auth.spec.ts:4]: I don't know why import statement changes every time.

File: packages/server/test/auth.spec.ts [hunk 4]
```diff
@@ -272,6 +276,28 @@ describe('test receive info', () => {
...
+        name: 'Kim',
+        studentId: '202002001',
+      },
+      {
+        authType: 'PACKAGE',
+        keyType: 'ECDSA',
+        value: '802A2697C771E2FC18C3E60074AEAFD84531783384EA3637F1EAFD9DFBF03885A21F1E1411447581E3618C0AD85D788B1DC6C348695977DD1C8EA1C7C8D04103',
+      },
+    ]);
+  });
```
Comment (HyeockJinKim) [inline packages/server/test/auth.spec.ts:300]: Scenario for verifying age and issuing student ID.

**Review by HyeockJinKim — COMMENTED** [review]
self review.

**Review by anthonyminyungi — APPROVED** [review]
Looks good. Sorry for the late response.

**Pull request merged**